### ISSUE TITLE
TypeError when using a JSON column in a many to many relationship

### ISSUE THREAD

**Issue by dees040** [laravel/nova-issues#3660]
- Laravel Version: 8.73.2
- Nova Version: 3.30.0
- PHP Version: 8.0.2
- Database Driver & Version: MySQL 8.0
- Operating System and Version: MacOS
- Browser type and version: Chrome

### Description:
I've a morph to many relationship between two model using an intermediate pivot model (`MorphPivot`). On the pivot model I've a JSON column which is also specified in `$casts` property (with the `array` type). In my use case it is possible that a model can have duplicate relations in the pivot table. So on the `MorphToMany` field I call `allowDuplicateRelations()`. Now when I try attach a resource I get the following error:

```
strtolower(): Argument #1 ($string) must be of type string, array given at /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php:836)
[stacktrace]
#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(836): strtolower(Array)
....
#7 /var/www/html/vendor/laravel/nova/src/Rules/NotExactlyAttached.php(63): Illuminate\\Database\\Eloquent\\Relations\\BelongsToMany->wherePivot('data', Array)
```

So in the `NotExactlyAttached` the `wherePivot()` method is being called using an array.

### Detailed steps to reproduce the issue on a fresh Nova installation:

1. Create a morph to many relationship using an intermediate pivot model.
2. In your Nova resource add a `MorphToMany` field with custom pivot fields.
3. To generate an array in those pivot fields I did use a package for a JSON column ([reference](https://github.com/64robots/nova-fields)).
4. Add the `allowDuplicateRelations()` to the `MorphToMany` field.
5. In Nova now attach a resource and the exception will be thrown.

**Comment by crynobone** [comment]
Can you provide a reproduction repository for us to understand and debug the issue.

**Comment by dees040** [comment]
[Here you go](https://github.com/dees040/nova-issues). After running `php artisan migrate:fresh --seed` you can go to `/nova/resources/posts/1/attach/tags?viaRelationship=tags&polymorphic=1` and try to attach a tag for the exception to be thrown.

**Comment by crynobone** [comment]
Fixed in v3.31.0: https://nova.laravel.com/releases/3.31.0
